syz-cluster/pkg/workflow/service.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package workflow
+
+import (
+	"sync"
+	"time"
+)
+
+// It's assumed that the workflow will query the necessary data and report its detailed progress itself,
+// so we only need to be able to start it and to check whether its current overall state.
+
+type Service interface {
+	Start(sessionID string) error
+	Status(id string) (Status, []byte, error)
+	// The recommended value. May depend on the implementation (test/prod).
+	PollPeriod() time.Duration
+}
+
+type Status string
+
+const (
+	StatusNotFound Status = "not_found"
+	StatusRunning  Status = "running"
+	StatusFinished Status = "finished"
+	StatusFailed   Status = "failed"
+)
+
+// MockService serializes callback invocations to simplify test implementations.
+type MockService struct {
```

Comment:
we have mockery

Reply:
Added to https://github.com/google/syzkaller/issues/5672